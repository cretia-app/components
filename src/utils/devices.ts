export const sizes = {
	mobileS: '320px',
	mobileM: '375px',
	mobileL: '425px',
	tablet: '768px',
	laptop: '1024px',
	laptopL: '1440px',
	desktop: '2560px',
} as const

export const TABLET_PORTRAIT_MIN_WIDTH = 768

export const devices = {
	mobileS: `(min-width: ${sizes.mobileS})`,
	mobileM: `(min-width: ${sizes.mobileM})`,
	mobileL: `(min-width: ${sizes.mobileL})`,
	tablet: `(min-width: ${sizes.tablet})`,
	laptop: `(min-width: ${sizes.laptop})`,
	laptopL: `(min-width: ${sizes.laptopL})`,
	desktop: `(min-width: ${sizes.desktop})`,
	desktopL: `(min-width: ${sizes.desktop})`,
} as const
