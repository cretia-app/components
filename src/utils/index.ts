import { useState } from 'react'

export { default as devices } from './devices'

export function useToggle(
	initialState = false,
): [boolean, (forceValue?: any) => void] {
	const [state, setState] = useState<boolean>(initialState)

	function toggleState(forceValue?: boolean) {
		if (typeof forceValue === 'boolean') {
			setState(forceValue)
		} else {
			setState((prev) => !prev)
		}
	}

	return [state, toggleState]
}
