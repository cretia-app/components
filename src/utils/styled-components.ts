import styledDefault, { css } from 'styled-components'

const styled: typeof styledDefault =
	(styledDefault as any).default || styledDefault

export { styled, css }
