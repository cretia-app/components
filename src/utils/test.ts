import { cleanup, render } from '@testing-library/react'
import { afterEach } from 'vitest'

afterEach(() => {
	cleanup()
})

// vi.mock('lottie-react', () => ({ default: () => null }))

const customRender = (ui: React.ReactElement, options = {}) =>
	render(ui, {
		...options,
	})

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
