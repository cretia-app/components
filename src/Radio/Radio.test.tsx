import { describe, expect, it, vi } from 'vitest'

import { render, screen } from '../__tests__/utils'

import { Radio } from '.'

const TEST_ID = 'radio-input'

describe.concurrent('<Radio />', () => {
	it('renders correctly', () => {
		const tree = render(
			<Radio
				onChange={() => undefined}
				value="value"
				name="radio"
				label="label"
				checked
			/>,
		)

		expect(tree).toMatchSnapshot()
	})

	it.concurrent('fires onChange on click', () => {
		const handleChange = vi.fn()

		render(
			<Radio
				value={1}
				label="Are you human?"
				onChange={handleChange}
				name="radio"
				checked={false}
			/>,
		)

		const radioInput = screen.getByTestId(TEST_ID) as HTMLInputElement

		expect(radioInput.checked).toBe(false)
	})
})
