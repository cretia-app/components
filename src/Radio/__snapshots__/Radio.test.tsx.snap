// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Radio /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="styles__Label-sc-a5hnpd-0 ivFQfl"
        data-testid="radio-label"
        for="radio-value"
      >
        label
        <input
          data-testid="radio-input"
          id="radio-value"
          name="radio"
          type="radio"
          value="value"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="styles__Label-sc-a5hnpd-0 ivFQfl"
      data-testid="radio-label"
      for="radio-value"
    >
      label
      <input
        data-testid="radio-input"
        id="radio-value"
        name="radio"
        type="radio"
        value="value"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
